--- old/modules/javafx.graphics/src/main/native-glass/win/Robot.cpp	2018-01-01 14:51:38.747630500 -0700
+++ new/modules/javafx.graphics/src/main/native-glass/win/Robot.cpp	2018-01-01 14:51:38.383610000 -0700
@@ -28,7 +28,6 @@
 
 #include "KeyTable.h"
 
-#include "com_sun_glass_ui_Robot.h"
 #include "com_sun_glass_ui_win_WinRobot.h"
 #include "GlassScreen.h"
 
@@ -294,10 +293,10 @@
 
 /*
  * Class:     com_sun_glass_ui_win_WinRobot
- * Method:    _getScreenCapture
+ * Method:    getScreenCapture
  * Signature: (IIII[I;)V
  */
-JNIEXPORT void JNICALL Java_com_sun_glass_ui_win_WinRobot__1getScreenCapture
+JNIEXPORT void JNICALL Java_com_sun_glass_ui_win_WinRobot_getScreenCapture
     (JNIEnv *env, jobject jrobot, jint x, jint y, jint width, jint height, jintArray pixelArray)
 {
     int numPixels = width * height;
