--- old/modules/javafx.graphics/src/main/java/module-info.java	2018-01-01 14:51:33.361320800 -0700
+++ new/modules/javafx.graphics/src/main/java/module-info.java	2018-01-01 14:51:32.990300700 -0700
@@ -53,6 +53,7 @@
     exports javafx.scene.input;
     exports javafx.scene.layout;
     exports javafx.scene.paint;
+    exports javafx.scene.robot;
     exports javafx.scene.shape;
     exports javafx.scene.text;
     exports javafx.scene.transform;
