--- old/modules/javafx.graphics/src/main/java/com/sun/glass/ui/gtk/GtkApplication.java	2018-01-01 14:51:12.993150100 -0700
+++ new/modules/javafx.graphics/src/main/java/com/sun/glass/ui/gtk/GtkApplication.java	2018-01-01 14:51:12.627129700 -0700
@@ -30,7 +30,6 @@
 import com.sun.glass.ui.Cursor;
 import com.sun.glass.ui.InvokeLaterDispatcher;
 import com.sun.glass.ui.Pixels;
-import com.sun.glass.ui.Robot;
 import com.sun.glass.ui.Screen;
 import com.sun.glass.ui.Size;
 import com.sun.glass.ui.Timer;
@@ -51,6 +50,8 @@
 import java.util.concurrent.CountDownLatch;
 import java.lang.annotation.Native;
 
+import javafx.scene.robot.Robot;
+
 final class GtkApplication extends Application implements
                                     InvokeLaterDispatcher.InvokeLaterSubmitter {
     private static final String SWT_INTERNAL_CLASS =
