--- old/modules/javafx.graphics/src/main/java/javafx/application/Application.java	2018-01-01 14:51:31.465212500 -0700
+++ new/modules/javafx.graphics/src/main/java/javafx/application/Application.java	2018-01-01 14:51:31.082190100 -0700
@@ -25,19 +25,17 @@
 
 package javafx.application;
 
-import java.security.AccessController;
-import java.security.PrivilegedAction;
 import java.util.List;
 import java.util.Map;
 
 import javafx.application.Preloader.PreloaderNotification;
 import javafx.scene.Scene;
+import javafx.scene.robot.Robot;
 import javafx.stage.Stage;
 
 import com.sun.javafx.application.LauncherImpl;
 import com.sun.javafx.application.ParametersImpl;
 import com.sun.javafx.application.PlatformImpl;
-import com.sun.javafx.css.StyleManager;
 
 /**
  * Application class from which JavaFX applications extend.
@@ -525,4 +523,15 @@
             PlatformImpl.setPlatformUserAgentStylesheet(url);
         }
     }
+
+    /**
+     * Creates and returns a {@link javafx.scene.robot.Robot} which can be used for
+     * simulating user interaction.
+     *
+     * @return a new {@link javafx.scene.robot.Robot} instance
+     * @since 11
+     */
+    public static Robot createRobot() {
+        return com.sun.glass.ui.Application.GetApplication().createRobot();
+    }
 }
