--- old/apps/tests/HelloTest/src/hellotest/RemoveFocusedControl.java	2018-01-01 14:51:03.695616500 -0700
+++ new/apps/tests/HelloTest/src/hellotest/RemoveFocusedControl.java	2018-01-01 14:51:03.314594700 -0700
@@ -30,9 +30,9 @@
 import javafx.scene.Group;
 import javafx.scene.Scene;
 import javafx.scene.control.Button;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.robot.Robot;
 import javafx.stage.Stage;
-import com.sun.glass.events.KeyEvent;
-import com.sun.glass.ui.Robot;
 
 public class RemoveFocusedControl extends Application {
 
@@ -62,14 +62,14 @@
 
         button1.setOnAction((ActionEvent e) -> {
             root.getChildren().remove(button1);
-            robot.keyPress(KeyEvent.VK_T);
-            robot.keyRelease(KeyEvent.VK_T);
-            robot.keyPress(KeyEvent.VK_E);
-            robot.keyRelease(KeyEvent.VK_E);
-            robot.keyPress(KeyEvent.VK_S);
-            robot.keyRelease(KeyEvent.VK_S);
-            robot.keyPress(KeyEvent.VK_T);
-            robot.keyRelease(KeyEvent.VK_T);
+            robot.keyPress(KeyCode.T);
+            robot.keyRelease(KeyCode.T);
+            robot.keyPress(KeyCode.E);
+            robot.keyRelease(KeyCode.E);
+            robot.keyPress(KeyCode.S);
+            robot.keyRelease(KeyCode.S);
+            robot.keyPress(KeyCode.T);
+            robot.keyRelease(KeyCode.T);
         });
 
         root.getChildren().add(button1);
